tests/Schema/ComparatorTest.php

```diff
@@ -1228,14 +1228,6 @@ public static function getCompareColumnComments(): iterable
         ];
     }
 
-    public function testCompareCommentedTypes(): void
-    {
-        $column1 = new Column('foo', Type::getType(Types::ARRAY));
-        $column2 = new Column('foo', Type::getType(Types::OBJECT));
-
-        self::assertFalse($this->comparator->columnsEqual($column1, $column2));
-    }
-
```

Comment:
Sometimes these columns should be considered equal, and sometimes not, it depends on the platform. Locally, this test now fails for SQLServer.

Reply:
It looks like this test uses the platform-unaware comparator. Why does changing the platform-aware comparison logic fail this test on any platform? I believe the platform-unaware comparison behavior should be preserved as much as possible unless it's a bug.

Reply:
> Locally, this test now fails for SQLServer.

It's not a functional test. How can it fail for SQL Server?

Reply:
> Sometimes these columns should be considered equal, and sometimes not, it depends on the platform.

Why does it depend on the platform? None of the platforms support these types natively, so I'd expect all platforms to behave the same. Although I agree that from the schema comparison perspective, these types should not be considered different (i.e. the test is invalid).

Reply:
For the above test to pass on SQL Server (which, ironically, isn't necessary after this patch applied), the platform-aware logic should have been implemented as:
```diff
diff --git a/src/Platforms/AbstractPlatform.php b/src/Platforms/AbstractPlatform.php
index 536b4a718..e51b40e5f 100644
--- a/src/Platforms/AbstractPlatform.php
+++ b/src/Platforms/AbstractPlatform.php
@@ -4017,9 +4017,12 @@ abstract class AbstractPlatform
      */
     public function columnsEqual(Column $column1, Column $column2): bool
     {
+        $column1Array = $this->columnToArray($column1);
+        $column2Array = $this->columnToArray($column2);
+
         if (
-            $this->getColumnDeclarationSQL('', $this->columnToArray($column1))
-            !== $this->getColumnDeclarationSQL('', $this->columnToArray($column2))
+            $this->getColumnDeclarationSQL('', $column1Array)
+            !== $this->getColumnDeclarationSQL('', $column2Array)
         ) {
             return false;
         }
@@ -4029,6 +4032,6 @@ abstract class AbstractPlatform
             return true;
         }

-        return $column1->getComment() === $column2->getComment();
+        return $column1Array['comment'] === $column2Array['comment'];
     }
 }
```
It should have compared the comments that would include the column type, not only the user-defined comments.

Until we're keeping the type comments in place, we should keep this test in place and apply the above patch. We can remove this patch together with removing commented types.

Reply:
Thanks for looking into this. I applied the suggested patch before proceeding with the loosening of the comparison so that each commit stays green.

Reply:
> I applied the suggested patch before proceeding with the loosening of the comparison so that each commit stays green.

I just realized that the end result is the same: instead of comparing comments and types separately, the comparator compares them in a combination. So there's no loosening of the column comparison logic in fact.

Reply:
Oh, right. Let's close this then.